src/components/ui/formik/RadioField.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { PropsWithoutRef } from 'react';
+import { FieldAttributes, useField } from 'formik';
+import { Radio, RadioProps } from '@patternfly/react-core';
+
+type RadioFieldProps = PropsWithoutRef<FieldAttributes<RadioProps>>;
+
+export const RadioField = (props: RadioFieldProps) => {
+  const [, , { setValue }] = useField({
+    name: props.name,
+    type: 'radio',
+  });
+
+  const handleChange = React.useCallback(
+    (checked, event) => {
+      const { value } = event.currentTarget;
+      if (checked) {
+        setValue(value === 'userManaged');
+      }
+    },
+    [setValue],
+  );
+
+  return <Radio {...props} onChange={handleChange} />;
```

Comment:
```suggestion
  return <Radio
      {...props}
      isChecked={
        (props.value === 'userManaged' && field.value) ||
        (props.value === 'clusterManaged' && !field.value)
      }
      onBlur={field.onBlur}
      onChange={handleChange}
    />;
```

Reply:
Note that the trick to avoid 'controlled -> uncontrolled' warnings was to use `isChecked` instead of `checked`